fox_scraper/fox_scraper.py

```diff
@@ -94,60 +94,66 @@ def verify_item_encoding(self, item_name):
-            items += self.get_sub_category_catalog(cat_sub_category, main_cat_num, sub_cat_num)
+        for sub_category in self.sub_categories_dict[main_category]:
+            cat_sub_category = self.get_parsed_web_content(sub_category["Sub-Category-Link"])
+            items += self.get_sub_category_catalog(cat_sub_category, main_category, sub_category["Sub-Category-Name"])
         return items
 
-    def get_sub_category_catalog(self, cat_sub_category, main_cat_num, sub_cat_num):
+    def get_sub_category_catalog(self, cat_sub_category, main_category, sub_category):
         """
         this function will receive web content for fox item and returns a list of items
+        :param main_category: main category name
+        :param sub_category: sub-category main category name
         :param cat_sub_category: web content for fox item
-        :param main_cat_num: main category number
-        :return:
+        :return: list of sub-catalog items
         """
         items = []
-        for link in cat_sub_category.find_all(class_='box padding_hf sp_p_padding_hf center border_b border_l margin_b_1'):
-            items.append(self.parse_item(link, main_cat_num, sub_cat_num))
+        for link in cat_sub_category.find_all(
+                class_='box padding_hf sp_p_padding_hf center border_b border_l margin_b_1'):
+            current_item = self.parse_item(link, main_category, sub_category)
+            items.append(current_item)
 
         return items
 
-    def parse_item(self, html_item, main_cat_num, sub_cat_num):
+    def parse_item(self, html_item, main_category, sub_category):
         """
         parse the information into Item class
         :return: Item class
         """
         item_img_id = html_item.find_next().get('href')
         item_name = html_item.find_next(class_='pname margin_hf_b').get_text()
-        item_price = html_item.find_next(class_='price display_inline numbers').find_next(class_='display_inline').get_text()
+        item_price = html_item.find_next(class_='price display_inline numbers').find_next(
+            class_='display_inline').get_text()
         nums = item_price.split()
         item_price = float(float(nums[0]) + (float(nums[1])) / 100)
-        item = FoxItem(item_img_id, self.main_categories[main_cat_num], self.sub_categories[main_cat_num][sub_cat_num], item_name, item_price)
+        item = FoxItem(item_img_id, main_category, sub_category,
+                       item_name, item_price)
         return item
 
-
-    # TODO - change location to new "program" function and not here
     def send_all_items(self, catalog):
+        """
+        send all catalog to rabbitmq
+        :param catalog: list of FoxItem's lists
+        :return:
+        """
+        with FoxSender() as rabbit_sender:
+            for items in catalog:
+                for item in items:
+                    rabbit_sender.send_message(item)
```

Comment:
or in one-liner 
`[rabbit_sender.send_message(item) for items in catalog for item in items]`